driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/ListSearchIndexesPublisherImpl.java

```diff
@@ -0,0 +1,131 @@
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.Document;
+import org.reactivestreams.Publisher;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+final class ListSearchIndexesPublisherImpl<T> extends BatchCursorPublisher<T> implements ListSearchIndexesPublisher<T> {
+    private Boolean allowDiskUse;
+    private long maxTimeMS;
+    private Collation collation;
+    private BsonValue comment;
+    private String indexName;
+
+    ListSearchIndexesPublisherImpl(
+            final MongoOperationPublisher<T> mongoOperationPublisher) {
+        super(null, mongoOperationPublisher);
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> name(@Nullable final String indexName) {
+        this.indexName = indexName;
+        return this;
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> allowDiskUse(@Nullable final Boolean allowDiskUse) {
+        this.allowDiskUse = allowDiskUse;
+        return this;
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> batchSize(final int batchSize) {
+        super.batchSize(batchSize);
+        return this;
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> maxTime(final long maxTime, final TimeUnit timeUnit) {
+        notNull("timeUnit", timeUnit);
+        this.maxTimeMS = TimeUnit.MILLISECONDS.convert(maxTime, timeUnit);
+        return this;
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> collation(@Nullable final Collation collation) {
+        this.collation = collation;
+        return this;
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> comment(@Nullable final String comment) {
+        this.comment = comment != null ? new BsonString(comment) : null;
+        return this;
+    }
+
+    @Override
+    public ListSearchIndexesPublisher<T> comment(@Nullable final BsonValue comment) {
+        this.comment = comment;
+        return this;
+    }
+
+    @Override
+    public Publisher<Document> explain() {
+        return publishExplain(Document.class, null);
+    }
+
+    @Override
+    public Publisher<Document> explain(final ExplainVerbosity verbosity) {
+        notNull("verbosity", verbosity);
+
+        return publishExplain(Document.class, verbosity);
+    }
+
+    @Override
+    public <E> Publisher<E> explain(final Class<E> explainResultClass) {
+        notNull("explainResultClass", explainResultClass);
+        return publishExplain(explainResultClass, null);
+    }
+
+    @Override
+    public <E> Publisher<E> explain(final Class<E> explainResultClass, final ExplainVerbosity verbosity) {
+        notNull("verbosity", verbosity);
+        return publishExplain(explainResultClass, verbosity);
+    }
+
+    private <E> Publisher<E> publishExplain(final Class<E> explainResultClass, @Nullable final ExplainVerbosity verbosity) {
+        notNull("explainDocumentClass", explainResultClass);
```

Comment:
This method can't be called by a user, which makes the `notNull` check inappropriate here.